From 0bf135257e67257be1c05b50169ebc63ec8cdde7 Mon Sep 17 00:00:00 2001
From: Gleb Mazovetskiy <glex.spb@gmail.com>
Date: Mon, 7 Nov 2022 02:46:30 +0000
Subject: [PATCH] Add an option to only build the tools

This is useful when cross-compiling.
Rather than building all of ecwolf, only configure and
build the `CROSS_EXPORTS` tools.
---
 CMakeLists.txt | 30 ++++++++++++++++++------------
 1 file changed, 18 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc38b47..0fffbb6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,6 +89,8 @@ elseif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 endif()
 
 # Support cross compiling
+option(TOOLS_ONLY "Only build the tools, not ecwolf itself (useful for cross-compiling)" OFF)
+
 option( FORCE_CROSSCOMPILE "Turn on cross compiling." NO )
 if( FORCE_CROSSCOMPILE )
 	set( CMAKE_CROSSCOMPILING TRUE )
@@ -175,17 +177,19 @@ else()
 	message(STATUS "Using system libbz2, includes found at ${BZIP2_INCLUDE_DIR}")
 endif()
 
-if(NOT JPEG_FOUND OR INTERNAL_JPEG)
-	message(STATUS "Using internal libjpeg")
-	set(JPEG_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/deps/jpeg-6b)
-	add_subdirectory(deps/jpeg-6b)
-	set(JPEG_LIBRARIES jpeg)
-	set(JPEG_LIBRARY jpeg)
-else()
-	message(STATUS "Using system libjpeg, includes found at ${JPEG_INCLUDE_DIR}")
-endif()
+if(NOT TOOLS_ONLY)
+	if(NOT JPEG_FOUND OR INTERNAL_JPEG)
+		message(STATUS "Using internal libjpeg")
+		set(JPEG_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/deps/jpeg-6b)
+		add_subdirectory(deps/jpeg-6b)
+		set(JPEG_LIBRARIES jpeg)
+		set(JPEG_LIBRARY jpeg)
+	else()
+		message(STATUS "Using system libjpeg, includes found at ${JPEG_INCLUDE_DIR}")
+	endif()
 
-include(LocateSDL2.cmake)
+	include(LocateSDL2.cmake)
+endif()
 
 if(MSVC)
 	add_definitions(-D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE)
@@ -210,8 +214,10 @@ if(ANDROID)
 	add_subdirectory(android-libs/launcher)
 endif()
 
-add_subdirectory(src)
-add_subdirectory(wadsrc)
+if(NOT TOOLS_ONLY)
+	add_subdirectory(src)
+	add_subdirectory(wadsrc)
+endif()
 
 if(USE_CPACK AND NOT APPLE)
 	if(WIN32)
-- 
2.37.2

